src/main/java/co/cask/hydrator/plugin/batch/HTTPSink.java

```diff
@@ -0,0 +1,402 @@
+                                                             HttpMethod.PUT, HttpMethod.DELETE);
+  private static StringBuilder messages = new StringBuilder("");
+
+  private HTTPSinkConfig config;
+
+  public HTTPSink(HTTPSinkConfig config) {
+    super(config);
+    this.config = config;
+  }
+
+  @Override
+  public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
+    super.configurePipeline(pipelineConfigurer);
+    config.validate();
+  }
+
+  @Override
+  public void prepareRun(BatchSinkContext context) throws Exception {
+    context.addOutput(Output.of(config.referenceName, new HTTPSink.HTTPOutputFormatProvider()));
+  }
+
+  @Override
+  public void transform(StructuredRecord input, Emitter<KeyValue<Void, Void>> emitter) throws Exception {
+    config.validate();
+    String message = null;
+    String contentType = null;
+    int responseCode;
+    int retries = 0;
+    Exception exception = null;
+
+    if (config.method.equals("POST") || config.method.equals("PUT")) {
+      if (config.messageFormat.equals("JSON")) {
+        message = StructuredRecordStringConverter.toJsonString(input);
+        contentType = "application/json";
+      } else if (config.messageFormat.equals("Form")) {
+        message = createFormMessage(input);
+        contentType = " application/x-www-form-urlencoded";
+      } else if (config.messageFormat.equals("Custom")) {
+        message = createCustomMessage(config.body, input);
+        contentType = "application/plain";
+      }
+      messages.append(message).append(config.delimiterForMessages);
+    }
+    StringTokenizer tokens = new StringTokenizer(messages.toString().trim(), config.delimiterForMessages);
+    if (config.batchSize == 1 || tokens.countTokens() == config.batchSize) {
+      do {
+        HttpURLConnection conn = null;
+        Map<String, String> headers = config.getRequestHeadersMap();
+        try {
+          URL url = new URL(config.url);
+          conn = (HttpURLConnection) url.openConnection();
+          if (conn instanceof HttpsURLConnection) {
+            //Disable SSLv3
+            System.setProperty("https.protocols", "TLSv1,TLSv1.1,TLSv1.2");
+            if (config.disableSSLValidation) {
+              disableSSLValidation();
+            }
+          }
+          conn.setRequestMethod(config.method.toUpperCase());
+          conn.setConnectTimeout(config.connectTimeout);
+          conn.setReadTimeout(config.readTimeout);
+          conn.setInstanceFollowRedirects(config.followRedirects);
+          conn.addRequestProperty("charset", config.charset);
+          for (Map.Entry<String, String> propertyEntry : headers.entrySet()) {
+            conn.addRequestProperty(propertyEntry.getKey(), propertyEntry.getValue());
+          }
+          if (config.method.equals("POST") || config.method.equals("PUT")) {
+            if (!headers.containsKey("Content-Type")) {
```

Comment:
If user has entered Content-Type in the headers then we are adding the same else the default one is being set for each message format.